doc/programmers/documentation.rst

```diff
@@ -37,12 +37,15 @@ appropriate ``.rst`` file.
 How does this work?
 -------------------
 
-To have an offline version of the documentation just issue ``make doc`` in the
-build directory.  The HTML will be stored in ``doc/html``. Open the
-``doc/html/index.html`` file with your browser to see and browse the
+To have an offline version of the documentation just issue 
+``sphinx-build doc/ _build/.``.  The HTML will be stored in ``_build/``. 
+Open the ``_build/index.html`` file with your browser to see and browse the
 documentation.
 
 .. warning::
 
-   Building the documentation requires Doxygen, Sphinx, Perl and the Python
-   modules PyYAML, Breathe and Matplotlib.
+   Building the documentation requires Python2, Doxygen, Sphinx, Perl and the 
```

Comment:
I think everything works with Python3 too. Could you check?

Reply:
`doc/conf.py` is not working with python3. I try it every time I forget to use *virtualenv*:).

I did a `2to3 -w doc/conf.py`, but then I got a strange error when I tried to compile the documentation with python3:
```
Exception occurred:
  File "/Volumes/repos/arnfinn/pcmsolver/doc/cloc_tools.py", line 152, in cloc_wrapper
    outfile.writelines(lines[6:])
TypeError: must be str, not bytes
```
Python2 worked fine.

Reply:
Can you grant me access to your fork? I'll try to make `conf.py` Python3 compatible

Reply:
I believe you have access? Can you try?

Reply:
Python3 _should_ work now. I say "should" because I can't get all the right packages on NixOS :disappointed:

Reply:
Tested it, and it seems to work. Unfortunately I get the following error when I try python2
```
Configuration error:
There is a syntax error in your configuration file: invalid syntax (conf.py, line 447)
```
which is 
```
[exec(compile(open(fname).read(), fname, 'exec')) for fname in list_of_scripts]
```

Maybe one of you python-gurus know how to make this line python2-likeable? It works with
```
[execfile(fname) for fname in list_of_scripts]
```
but then python3 fails.

Reply:
Oh no!!! I guess the guru is @bast in this case. That line executes the scripts generated to plot the bar charts. A better solution would be to `import` those files as they are generated and execute the function they define.

Reply:
Exec indeed always looks a bit dangerous. Do you want me to look at that?

Reply:
If it's quick (<= 30 minutes) business for you, then yes please.

Reply:
@bast I gave you access to my fork

Reply:
OK I shall have a quick look. Thanks for access but I would have simply forked your fork.

Reply:
This works:
```diff
diff --git a/doc/conf.py b/doc/conf.py
index aa8fe4a7..8b11f678 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -444,7 +444,8 @@ def generate_bar_charts(mod_dir, dir_lang, savedir):
     # Generate scripts and list of scripts (absolute paths)
     list_of_scripts = [bar_chart(root_dir, language, savedir) for root_dir, language in dir_lang.items()]
     # Generate charts
-    [exec(compile(open(fname).read(), fname, 'exec')) for fname in list_of_scripts]
+    for fname in list_of_scripts:
+        exec(compile(open(fname).read(), fname, 'exec'))
```

Importing things would be cleaner but this is more work since the scripts that are executed there have everything in main scope.

Reply:
@bast Thanks!